tests/unit/s2n_client_hello_get_supported_groups_test.c

```diff
@@ -0,0 +1,400 @@
+        } else {
+            EXPECT_SUCCESS(ret);
+
+            /* The 20170210 security policy contains 2 ECC curves. */
+            EXPECT_EQUAL(supported_groups_count, 2);
+        }
+    }
+
+    /* Test parsing a supported groups extension with a malformed groups list length. */
+    {
+        struct s2n_client_hello client_hello = { 0 };
+
+        s2n_extension_type_id supported_groups_id = 0;
+        EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(S2N_EXTENSION_SUPPORTED_GROUPS, &supported_groups_id));
+
+        s2n_parsed_extension *supported_groups_extension = &client_hello.extensions.parsed_extensions[supported_groups_id];
+        supported_groups_extension->extension_type = S2N_EXTENSION_SUPPORTED_GROUPS;
+
+        /* Test parsing a correct groups list length */
+        {
+            uint8_t extension_data[S2N_TEST_SUPPORTED_GROUPS_EXTENSION_SIZE] = { 0 };
+            struct s2n_blob extension_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&extension_blob, extension_data, sizeof(extension_data)));
+            supported_groups_extension->extension = extension_blob;
+
+            struct s2n_stuffer extension_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&extension_stuffer, &extension_blob));
+
+            EXPECT_SUCCESS(s2n_stuffer_write_uint16(&extension_stuffer, S2N_TEST_SUPPORTED_GROUPS_LIST_SIZE));
+
+            uint16_t supported_groups[S2N_TEST_SUPPORTED_GROUPS_LIST_COUNT] = { 0 };
+            uint16_t supported_groups_count = 0;
+            EXPECT_SUCCESS(s2n_client_hello_get_supported_groups(&client_hello, supported_groups,
+                    s2n_array_len(supported_groups), &supported_groups_count));
+
+            EXPECT_EQUAL(supported_groups_count, S2N_TEST_SUPPORTED_GROUPS_LIST_COUNT);
+        }
+
+        /* Test parsing a groups list length that is larger than the extension length */
+        {
+            uint8_t extension_data[S2N_TEST_SUPPORTED_GROUPS_EXTENSION_SIZE] = { 0 };
+            struct s2n_blob extension_blob = { 0 };
+            uint32_t extension_too_small_size = S2N_TEST_SUPPORTED_GROUPS_EXTENSION_SIZE - 2;
+            EXPECT_SUCCESS(s2n_blob_init(&extension_blob, extension_data, extension_too_small_size));
+            supported_groups_extension->extension = extension_blob;
+
+            struct s2n_stuffer extension_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&extension_stuffer, &extension_blob));
+            EXPECT_SUCCESS(s2n_stuffer_write_uint16(&extension_stuffer, S2N_TEST_SUPPORTED_GROUPS_LIST_SIZE));
+
+            uint16_t supported_groups[S2N_TEST_SUPPORTED_GROUPS_LIST_COUNT] = { 0 };
+            uint16_t supported_groups_count = 0;
+            EXPECT_FAILURE_WITH_ERRNO(s2n_client_hello_get_supported_groups(&client_hello, supported_groups,
+                                              s2n_array_len(supported_groups), &supported_groups_count),
+                    S2N_ERR_SAFETY);
```

Comment:
Is S2N_ERR_SAFETY right here? Isn't this an error we kind of expect / a customer might see? Same with the other ones related to a malformed extension

Reply:
I changed this to use the  existing`S2N_ERR_INVALID_PARSED_EXTENSIONS` error.